### ISSUE TITLE
[Bug]: supports_blit is True with MacOSX backend

### ISSUE THREAD

**Issue by cpmoni** [matplotlib/matplotlib#22690]
### Bug summary

Despite MacOSX not supporting blitting (see [here](https://matplotlib.org/3.5.1/tutorials/advanced/blitting.html)), figures created with the MacOSX backend still have `fig.canvas.supports_blit = True`.  However, running the example in the linked tutorial does not produce the expected output as blitting is not supported.

### Code for reproduction

```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots()
print(fig.canvas.supports_blit)
```

### Actual outcome

True
![Screen Shot 2022-03-23 at 10 30 42 AM](https://user-images.githubusercontent.com/12058086/159749146-1aa443fc-19d1-4e79-bb1f-779afb5c66a7.png)

### Expected outcome

False

### Operating system

Mac OS 12.3

### Matplotlib Version

3.5.1

### Matplotlib Backend

MacOSX

### Python version

3.8.3

### Installation

pip

**Comment by tacaswell** [comment]
I think this should work now (4264233abf9e4c71e6d1cb93bcc36a81bf1fa1aa).  Does it not actually work or do we need to update the docs?

**Comment by greglucas** [comment]
It should work on **main**, but that update wasn't backported to 3.5.x. I guess we could think about backporting that PR, but there are a lot of updates coming in 3.6 for the macosx backend, so I'd hesitate to only backport this one fix without many of the others too because they may all have some interplay to a certain extent.

Alternatively, we could update 3.5.x to say that the backend doesn't support blitting, but it has been that way for so long that I'm not sure it is a great idea to change that just for a bugfix release because someone could be relying on that flag for something else.

Unfortunately, I think that means we just have to live with it in this inconsistent state until 3.6 is released :-/ I'm not tied to any of these ideas though, so if someone else feels strongly about either of those proposals we could definitely tackle them.

**Comment by QuLogic** [comment]
I guess this was fixed with 3.6 and can be closed.
